apache2/re.c

```diff
@@ -252,22 +252,21 @@ char *update_rule_target_ex(modsec_rec *msr, msre_ruleset *ruleset, msre_rule *r
     if(rule != NULL)    {
 
         target_list = strdup(p2);
-        if(target_list == NULL)
-            return apr_psprintf(ruleset->mp, "Error to update target - memory allocation");;
+        if (target_list == NULL) {
+            my_error_msg = apr_psprintf(ruleset->mp, "Error to update target - memory allocation");
```

Comment:
Here (and at some other cases) you removed the `return` statement and after set of `my_error_msg` you jump to the label `end`. There you call an `msr_log(msr, 9, my_error_msg)` and always return with NULL.

But the calling places return this value, so may be that `char *` needs there - isn't it?

Reply:
You're right, I fixed the return value